src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -1001,13 +1001,10 @@ public static int getNumberOfChunks(long size) {
         LOG.info("Overwriting section [" + Long.toHexString(inputStartIndex) + ", " + Long.toHexString(endIndex) + "] of child with name: " + filename);
 
         return current.withWriter(existingChild.owner(), existingChild.writer(), network)
-                .thenCompose(state -> (existingChild.isDirty() ?
+                .thenCompose(state ->
                         existingChild.clean(state, committer, network, crypto, parent)
-                                .thenCompose(pair -> pair.left.getChild(pair.right, filename, crypto.hasher, network)
-                                        .thenApply(cleanedChild -> cleanedChild.get())) :
-                        CompletableFuture.completedFuture(existingChild))
-                ).thenCompose(updatedChild -> updatedChild.overwriteSection(updatedChild.version, committer, fileData,
-                        inputStartIndex, endIndex, network, crypto, monitor));
+                                .thenCompose(pair -> pair.left.overwriteSection(pair.right, committer, fileData,
```

Comment:
What was wrong here?

Reply:
previously clean() returned Pair<parent, Snapshot> (although it didn't do that in all cases). Now that an updated ref to the FW is returned, there is no reason to go from parent to child.